### PR TITLE
interfaces/display-control: allow changing brightness value

### PR DESCRIPTION
**Pull request by bugraaydogar**
Once a developer would like to change the brigness of the display, they get the following AppArmor denials. This PR aims to fix that.

```
apparmor="DENIED" operation="open" profile="snap.my-app.awesome-ui" name="/sys/devices/pci0000:00/0000:00:02.0/drm/card0/card0-eDP-1/intel_backlight/brightness" pid=1859 comm="bash" requested_mask="wc" denied_mask="wc" fsuid=0 ouid=0 

```

### CODE REVIEW

**Comment by MiguelPires** [comment]
Thanks for this. I added the "Needs security review" label since it's an interface change

File: interfaces/builtin/display_control.go
Comment (bugraaydogar) [reply interfaces/builtin/display_control.go:88]: can we? It is a symlink so I was not sure.
```
lrwxrwxrwx  1 root root 0 Haz 30 11:59 intel_backlight -> ../../devices/pci0000:00/0000:00:02.0/drm/card0/card0-eDP-1/intel_backlight
```

Comment (mardy) [reply interfaces/builtin/display_control.go:88]: For symbolic links, we need to allow both paths (symbolic link name and target) in order to have it working.

IMHO it makes sense to list the symlink too.

About the "target" rule, I guess it should be either
```suggestion
/sys/devices/**/**/drm/card[0-9]/**_backlight/brightness, w
```
or
```suggestion
/sys/devices/**/**/drm/card[0-9]/card[0-9]*/*_backlight/brightness, w
```

**Review by mardy — COMMENTED** [review]
I'm unsure about granting direct access, since we already have rules for GNOME and KDE; but indeed, in those systems (like core) where neither is present, we have no alternative to a direct file access.

I wonder, though, if there should be an interface attribute specifying a different flavor? Maybe I'm overthinking it, since AFAIK the GNOME and KDE interfaces are just convinence APIs and I guess nothing bad happens if one plays with the brightness settings directly...

I think the current approach is fine, I just raise the point in case someone sees any risk there.

File: interfaces/builtin/display_control.go
Comment (bugraaydogar) [reply interfaces/builtin/display_control.go]: Thanks, updated after the comments.

**Review by mardy — APPROVED** [review]
LGTM, but let's wait for the security review before merging this.

**Review by alexmurray — CHANGES_REQUESTED** [review]
I don't think the symlink name itself is needed - just the target - since AppArmor rules only use the target - as such I think the `/sys/class/backlight/...` rule should be removed for clarity.

**Comment by bugraaydogar** [comment]
The PR is modified as per @alexmurray comment.

**Review by MiguelPires — APPROVED** [review]
LGTM

**Review by alexmurray — APPROVED** [review]
LGTM!

File: interfaces/builtin/display_control.go
Comment (MiguelPires) [inline interfaces/builtin/display_control.go:88]:
```suggestion
/sys/devices/**/**/drm/card[0-9]/card[0-9]*/*_backlight/brightness w,
```

**Review by MiguelPires — COMMENTED** [review]
The comma is misplaced

File: interfaces/builtin/display_control.go
Comment (bugraaydogar) [reply interfaces/builtin/display_control.go]: Thanks Miguel, very good catch.

**Review by MiguelPires — APPROVED** [review]
Thank you

**Pull request merged**
